src/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -86,6 +85,25 @@ app.directive('dashboardApp', function ($injector) {
         dashboardStateManager.syncTimefilterWithDashboard(timefilter, quickRanges);
       }
 
+      let onRouteChange;
+      const setFullScreenMode = fullScreenMode => {
+        if (dashboardStateManager.getFullScreenMode() !== fullScreenMode) {
+          dashboardStateManager.setFullScreenMode(fullScreenMode);
```

Comment:
Just moved this chunk of code up to avoid the "used before defined" lint error, but did make a change here, now that updates can go both ways (angular -> redux and redux -> angular) need this to avoid an infinite loop.